apollo-mpp-utils/src/appleMain/kotlin/com/apollographql/apollo3/mpp/utils.kt

```diff
@@ -11,10 +13,24 @@ actual fun currentTimeMillis(): Long {
   return getTimeMillis()
 }
 
+private val nsDateFormatter by lazy { NSDateFormatter().apply { dateFormat = "HH:mm:ss.SSS" } }
+
+actual fun currentTimeFormatted(): String {
+  return nsDateFormatter.stringFromDate(NSDate())
+}
+
 actual fun currentThreadId(): String {
   return pthread_self()?.rawValue.toString()
 }
 
+actual fun currentThreadName(): String {
+  return if (NSThread.isMainThread) {
+    "main"
+  } else {
+    currentThreadId()
```

Comment:
Could you use [pthread_get_name_np](https://linux.die.net/man/3/pthread_getname_np) instead? Does iOS put anything relevant there?

Reply:
Sadly, this returns an empty string (same for `NSThread.currentThread.name`)!